modules/dkan/dkan_datastore/modules/dkan_datastore_api/dkan_datastore_api.module

```diff
@@ -1123,3 +1123,180 @@ function dkan_datastore_api_debug($data_select) {
   $sql = str_replace('\n', ' ', $sql);
   return (string) $sql;
 }
+
+/**
+ * Implements hook_services_resources_alter().
+ */
+function dkan_datastore_api_services_resources_alter(&$resources, $endpoint) {
+  if (isset($resources['node']['operations']['create'])) {
+    $resources['node']['operations']['create']['callback'] =
+      "_dkan_datastore_api_resource_create";
+  }
+
+  if (isset($resources['node']['operations']['update'])) {
+    $resources['node']['operations']['update']['callback'] =
+      "_dkan_datastore_api_resource_update";
+  }
+}
+
+/**
+ * Services operation callback: node creation.
+ */
+function _dkan_datastore_api_resource_create($request) {
+  $node_array = _services_arg_value($request, 'node');
+  $node = (object) $node_array;
+
+  if (isset($node->type)) {
+    if ($node->type == "dataset") {
+
+      try {
+        field_attach_validate('node', $node);
+        _dkan_datastore_api_field_validation($node);
+      }
+      catch (\Exception $e) {
+        return services_error($e->getMessage());
+      }
+
+      node_save($node);
+
+      $nid = $node->nid;
+      $response = array('nid' => $nid);
+      if ($uri = services_resource_uri(array('node', $nid))) {
+        $response['uri'] = $uri;
+      }
+    }
+    else {
+      $response = _node_resource_create($request);
+    }
+  }
+  else {
+    return services_error("Type not set for node");
+  }
+
+  return $response;
+}
+
+/**
+ * Services operation callback: node update.
+ */
+function _dkan_datastore_api_resource_update($nid, $request) {
+  $old_node = node_load($nid);
+
+  if (isset($old_node->type)) {
+    if ($old_node->type == "dataset") {
+
+      if (empty($old_node->nid)) {
+        return services_error(t('Node @nid not found', array('@nid' => $old_node->nid)), 404);
+      }
+
+      $node = _services_arg_value($request, 'node');
+
+      if (!empty($node['type']) && $node['type'] != $old_node->type) {
+        // Node types cannot be changed once they are created.
+        return services_error(t('Node type cannot be changed'), 406);
+      }
+
+      foreach ($node as $field => $value) {
+        $old_node->{$field} = $value;
+      }
+
+      try {
+        field_attach_validate('node', $old_node);
+        _dkan_datastore_api_field_validation($old_node);
+      }
+      catch (\Exception $e) {
+        return services_error($e->getMessage());
+      }
+
+      node_save($old_node);
+
+      $nid = $old_node->nid;
+      $response = array('nid' => $nid);
+      if ($uri = services_resource_uri(array('node', $nid))) {
+        $response['uri'] = $uri;
+      }
+      return $response;
+    }
+    else {
+      return _node_resource_update($nid, $request);
```

Comment:
What function is this? `_node_resource_update()`

Reply:
This function is the original callback for all node related service calls, since we are handling only a few node types we want to pass the rest of the calls to services.